deployment/data-streams/changeset/call_configurator_set_production_config.go

```diff
@@ -0,0 +1,80 @@
+package changeset
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	ethTypes "github.com/ethereum/go-ethereum/core/types"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/llo-feeds/generated/configurator"
+)
+
+// SetProductionConfigConfig contains the parameters needed to set a production config.
+type SetProductionConfigConfig struct {
+	// ConfigurationsByChain maps chain selectors to production configuration settings.
+	ConfigurationsByChain map[uint64][]SetProductionConfig
+	MCMSConfig            *MCMSConfig
+}
+
+// SetProductionConfig groups the parameters for a production config call.
+type SetProductionConfig struct {
+	ConfiguratorAddress   common.Address
+	ConfigID              [32]byte
+	Signers               [][]byte
+	OffchainTransmitters  [][32]byte
+	F                     uint8
+	OnchainConfig         []byte
+	OffchainConfigVersion uint64
+	OffchainConfig        []byte
+}
+
+func (pc SetProductionConfig) GetConfiguratorAddress() common.Address {
+	return pc.ConfiguratorAddress
+}
+
+// Validate checks the production configuration.
+func (cfg SetProductionConfigConfig) Validate() error {
+	if len(cfg.ConfigurationsByChain) == 0 {
+		return errors.New("ConfigurationsByChain cannot be empty")
+	}
+	return nil
+}
+
+func getTransactOpts(e deployment.Environment, chainSel uint64, mcmsConfig *MCMSConfig) *bind.TransactOpts {
+	if mcmsConfig == nil {
+		return e.Chains[chainSel].DeployerKey
+	}
+	return deployment.SimTransactOpts()
+}
+
+func setProductionConfigTx(
+	e deployment.Environment,
+	configuratorContract *configurator.Configurator,
+	prodCfg SetProductionConfig,
+	opts *bind.TransactOpts,
+	chain deployment.Chain,
+	mcmsConfig *MCMSConfig,
+) (*ethTypes.Transaction, error) {
+	tx, err := configuratorContract.SetProductionConfig(opts, prodCfg.ConfigID, prodCfg.Signers, prodCfg.OffchainTransmitters, prodCfg.F, prodCfg.OnchainConfig, prodCfg.OffchainConfigVersion, prodCfg.OffchainConfig)
+	if err != nil {
+		return nil, fmt.Errorf("error packing setProductionConfig tx data: %w", err)
+	}
+
+	if mcmsConfig == nil {
+		if _, err := deployment.ConfirmIfNoError(chain, tx, err); err != nil {
+			e.Logger.Errorw("Failed to confirm setProductionConfig tx", "chain", chain.String(), "err", err)
+			return nil, err
+		}
+	}
+	return tx, nil
+}
+
+func CallSetProductionConfig(e deployment.Environment, cfg SetProductionConfigConfig) (deployment.ChangesetOutput, error) {
```

Comment:
This looks like it's matching the deprecated ChangeSet approach - can we migrate these to V2?

Reply:
Done!